src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Extensibility;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    // General binder that works directly off SDK interfaces. 
+    class GeneralScriptBindingProvider : ScriptBindingProvider
+    {
+        public ITooling Tooling { get; set; }
+
+        public GeneralScriptBindingProvider(
+            JobHostConfiguration config, 
+            JObject hostMetadata, 
+            TraceWriter traceWriter) :
+            base(config, hostMetadata, traceWriter)
+        {
+        }
+
+        public override bool TryCreate(ScriptBindingContext context, out ScriptBinding binding)
+        {
+            string name = context.Type;
+            var attrType = this.Tooling.GetAttributeTypeFromName(name);
+            if (attrType == null)
+            {
+                binding = null;
+                return false;
+            }
+
+            var attrs = this.Tooling.GetAttributes(attrType, context.Metadata);
+                        
+            binding = new GeneralScriptBinding(this.Tooling, attrs, context);
+            return true;
+        }
+
+        public override bool TryResolveAssembly(string assemblyName, out Assembly assembly)
+        {
+            assembly = this.Tooling.TryResolveAssembly(assemblyName);
```

Comment:
maybe change the Tooling interface to be `bool TryResolveAssembly(name, out Assembly)` so that it matches this method & the `Try` semantics better